daml-assistant/src/DAML/Assistant/Install/Github.hs

```diff
@@ -83,41 +42,38 @@ tagToVersion (Tag t) =
         else
             Left "Tag must start with v followed by semantic version."
 
--- | Make a request to the Github API. There is an unauthenticated user rate limit
--- of 60 requests per hour, so we should limit requests. (One way for the user to
--- avoid API requests is to give the version directly.)
-makeAPIRequest :: FromJSON t => Text -> IO t
-makeAPIRequest path = do
-    request <- parseRequest (unpack ("GET https://api.github.com/repos/digital-asset/daml" <> path))
-    response <- httpJSON (setRequestHeader "User-Agent" ["daml"] request)
-    when (getResponseStatusCode response /= 200) $
-        throwString . show $ getResponseStatus response
-    pure (getResponseBody response)
+-- | Get the version of the latest stable (i.e. non-prerelease) release.
+-- We avoid the Github API because of very low rate limits. As such, we
+-- discover the latest version by parsing an HTTP redirect. We make a
+-- request to:
+--
+--     https://github.com/digital-asset/daml/releases/latest
+--
+-- Which always redirects to the latest stable release, for example:
+--
+--     https://github.com/digital-asset/daml/releases/tag/v0.12.3
+--
+-- So we take that URL to get the tag, and from there the version of
+-- the latest stable release.
+getLatestVersion ::  IO SdkVersion
+getLatestVersion = do
+
+    manager <- newTlsManager -- TODO: share a single manager throughout the daml install process.
+    request <- parseRequest "HEAD https://github.com/digital-asset/daml/releases/latest"
```

Comment:
No way to avoid parsing HEAD from a string? A bit sad, but I see from the docs it's recommended...